Updated the Terra guide (#1298)
DataBiosphere/data-portal: content/guides/userguides/consumer-vignettes/export-to-terra.md

title: "Exporting HCA Data to Terra"
---

# Analyzing HCA Data in Terra

> [Terra](https://app.terra.bio/) is a scalable cloud platform for biomedical research that offers the ability to use data, tools, and workflows to do interactive analysis in the cloud.

Reviewer: ```suggestion
[Terra](https://app.terra.bio/) is a scalable cloud platform for biomedical research that offers the ability to use data, tools, and workflows to perform interactive analysis in the cloud.
```

In this tutorial, you will learn how to send search results from the HCA Data Explorer to Terra and how to run a basic workflow with that data.

## Before starting

> Before trying the tutorial, complete the steps below:

Reviewer: ```suggestion
Before trying this tutorial, complete the steps below:
```

1. Read the [Accessing HCA Data and Metadata](../quick-start-guide) guide to learn how to find HCA data in the Data Explorer.
1. Visit [Terra Support](https://support.terra.bio/hc/en-us) to learn how to [register for a Terra account](https://support.terra.bio/hc/en-us/articles/360028235911).

> 1. Download the [Google Chrome browser](https://support.terra.bio/hc/en-us/articles/360028235911) which is recommended when using Terra.

Reviewer: Do you think this is necessary if it's mentioned in the article you linked right above? If you want to keep it, I think I would link to a page to actually download Google Chrome.

Author: It's come up enough that I'm fine with a little redundancy. I'll add the direct link to Chrome.

> 1. Download the [Google Chrome browser](https://support.terra.bio/hc/en-us/articles/360028235911) which is recommended when using Terra.

Author: ```suggestion
1. Download the [Google Chrome browser](https://www.google.com/chrome/) which is recommended when using Terra.
```

### HCA data-related tutorial workspaces
In addition to running the tutorial below, registering for Terra allows you to accesss multiple tutorial workspaces dedicated to using HCA data as described below:

* [Optimus workspace](https://app.terra.bio/#workspaces/featured-workspaces-hca/HCA_Optimus_Pipeline) for processing 10x data with the [Optimus Pipeline](/pipelines/optimus-workflow).
* [Smart-seq2 workspace](https://app.terra.bio/#workspaces/featured-workspaces-hca/HCA%20Smart-seq2%20Multi%20Sample%20Pipeline) for processing Smart-seq2 data with the [Smart-seq2 Multi-Sample Pipeline](/pipelines/smart-seq2-workflow).
* [Intro-to-HCA-data-on-Terra workspace](https://app.terra.bio/#workspaces/featured-workspaces-hca/Intro-to-HCA-data-on-Terra) for exporting HCA data and analyzing it with community tools like [Seurat](https://satijalab.org/seurat/index.html), [Scanpy](https://scanpy-tutorials.readthedocs.io/en/latest/index.html), [Cumulus](https://cumulus.readthedocs.io/en/latest/index.html), and [Pegasus](https://pegasus.readthedocs.io/en/stable/#). 

### Video demonstration of importing HCA data to a Terra workspace

> Prior to trying the tutorial, you might want to watch this very brief [video demonstration](https://www.youtube.com/watch?v=G3N2i3NjsfY) of importing HCA data into Terra and using the Intro-to-HCA-data-on-Terra workspace. This video is additionally available in this [Terra blog](https://terra.bio/discover-how-to-use-human-cell-atlas-data-in-terra/) about the workspace.

Reviewer: ```suggestion
Prior to trying the tutorial, you might want to watch this very brief [video demonstration](https://www.youtube.com/watch?v=G3N2i3NjsfY) of importing HCA data into Terra and using the Intro-to-HCA-data-on-Terra workspace. This video is additionally available in this [Terra blog post](https://terra.bio/discover-how-to-use-human-cell-atlas-data-in-terra/) about the workspace.
```

## Tutorial — Automated processing of HCA BAM files in Terra 

This tutorial guides you through finding and setting up a simple example workflow in Terra that you can use to process BAM files exported from the HCA Data Explorer.

> For an additional Terra tutorial on using community tools with HCA data, see the [Intro-to-HCA-data-on-Terra workspace](https://app.terra.bio/#workspaces/featured-workspaces-hca/Intro-to-HCA-data-on-Terra). 

Reviewer: I might put this at the end of the tutorial as an extra step to try if you want more practice kind of thing.

Author: Thanks for the suggestion. I added it to the end as well. I put it here too because I wanted them to see it before they started on this tutorial in case they are more interested in community tools than using the MD5sum workflow.

Using the Data Explorer, select some data that you are interested in. Choose anything that looks interesting - we will be running a really simple workflow that generates MD5 checksums of files, so the type of data is not important.

When you have found a data set of interest, use the following steps:
1. Click on the big blue *Export Selected Data* icon at the top right of the page. 
1. Click on the *Analyze in Terra* icon at the bottom.

	![](../../_images/analyze-in-terra-export.png)

1. Click on the species you want to analyze.
1. Select the files you want to export by clicking the checkbox next to the file type.
1. Select the download format (the terra.pfb option is in development).
1. Click the *Request Link* icon.
	
	You'll be redirected to a Data Portal page containing an export link.

1. Click the workspace URL link.
	
	You'll be redirected to the Terra import page.

1. If you've already created a Terra workspace, select workspace by choosing an existing workspace or choose "Start with a new workspace."

	![](../../_images/terra-import.png) 

> ### Step two: finding a workflow in Dockstore

Reviewer: It might be worth adding a link to a WDL resource somewhere in this section.

Author: Good idea! Added.

A workflow is simply a set of steps you want to use to transform or analyze your data, often written as a script in a language like Python or R. Terra integrates with scripts written in a particular workflow language called the Workflow Description Language (WDL).  

We can find WDL workflows that use our analysis tools of interest from public repositories like [Dockstore](https://dockstore.org/), a platform for bioscience tools and workflows that use Docker containers. For more information about how to use the Dockstore, see the [Dockstore documentation](https://docs.dockstore.org/en/develop/).

We'll use Dockstore in this tutorial to find a workflow to run on the data we've just exported. Specifically, we'll look for the *dockstore-wdl-workflow-md5sum* workflow, which generates an MD5 checksum for a given file (or files). 

We will need to import this workflow from Dockstore using the following steps: 

> 1. Select on the *Workflows* page at the top of the workspace.

Reviewer: ```suggestion
1. Select the *Workflows* page at the top of the workspace.
```
